krod/modules/research/web_search/search_providers.py

```
@@ -106,10 +102,81 @@ def search(self, query: str, num_results: int = 5) -> List[Dict[str, Any]]:
+            self.logger.error("Cannot search without API key")
+            return []
+        
+        headers = {
+            "Ocp-Apim-Subscription-Key": self.api_key
+        }
+
+        params = {
+            "q": query,
+            "count": num_results,
+            "responseFilter": "Webpages",
+            "textDecorations": false,
```

Ruff (0.11.9)

Undefined name `false`

(F821)